File: agentwallet/packages/sdk-python/src/agentwallet/resources/policies.py
"""Policies sub-resource."""

from __future__ import annotations

from typing import TYPE_CHECKING

from ..types import ListResponse, Policy

if TYPE_CHECKING:
    from ..client import AgentWallet


class PoliciesResource:
    def __init__(self, client: AgentWallet):
        self._client = client

    async def create(
        self,
        name: str,
        rules: dict,
        scope_type: str = "org",
        scope_id: str | None = None,
        priority: int = 100,
    ) -> Policy:
        data = await self._client.post("/policies", json={
            "name": name,
            "rules": rules,
            "scope_type": scope_type,
            "scope_id": scope_id,
            "priority": priority,
        })
        return Policy(**data)

    async def get(self, policy_id: str) -> Policy:
        data = await self._client.get(f"/policies/{policy_id}")
        return Policy(**data)

    async def list(self, limit: int = 50, offset: int = 0) -> ListResponse:
        data = await self._client.get("/policies", params={"limit": limit, "offset": offset})
        return ListResponse(
            data=[Policy(**p) for p in data["data"]],
            total=data["total"],
        )

    async def update(self, policy_id: str, **kwargs) -> Policy:
        data = await self._client.patch(f"/policies/{policy_id}", json=kwargs)
        return Policy(**data)

    async def delete(self, policy_id: str) -> None:
        await self._client.delete(f"/policies/{policy_id}")

File: agentwallet/packages/sdk-ts/src/resources/policies.ts
import type { AgentWallet } from '../client';
import type { CreatePolicyParams, ListResponse, Policy } from '../types';

export class PoliciesResource {
  constructor(private client: AgentWallet) {}

  /** Create a spending policy */
  async create(params: CreatePolicyParams): Promise<Policy> {
    return this.client.post<Policy>('/policies', {
      name: params.name,
      rules: params.rules,
      scope_type: params.scope_type ?? 'org',
      scope_id: params.scope_id ?? null,
      priority: params.priority ?? 100,
    });
  }

  /** Get a policy by ID */
  async get(policyId: string): Promise<Policy> {
    return this.client.get<Policy>(`/policies/${policyId}`);
  }

  /** List policies */
  async list(options?: { limit?: number; offset?: number }): Promise<ListResponse<Policy>> {
    return this.client.get<ListResponse<Policy>>('/policies', {
      limit: options?.limit ?? 50,
      offset: options?.offset ?? 0,
    });
  }

  /** Update a policy */
  async update(policyId: string, params: Partial<CreatePolicyParams>): Promise<Policy> {
    return this.client.patch<Policy>(`/policies/${policyId}`, params as Record<string, unknown>);
  }

  /** Delete a policy */
  async delete(policyId: string): Promise<void> {
    await this.client.delete(`/policies/${policyId}`);
  }
}
